File: HashTable/219. Contains Duplicate II.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        //initialize a hash map
        Map<Integer, Integer> hashmap = new HashMap<>();

        for (int index = 0; index < nums.length; index++) {
            //check if it's already there
            if(!hashmap.containsKey(nums[index])){
                hashmap.put(nums[index], index);
            } // add the index value at the spot
            else {
                if(-k <= (hashmap.get(nums[index]) -index) ) return true;
                hashmap.put(nums[index], index);
            }
        }
        return false;
        
    }
}

File: HashTable/219. Contains Duplicate II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        dict = {} #hash map

        index = 0
        while(index < len(nums)):
            if(nums[index] not in dict):
                dict[nums[index]] =index #keep track of the indexes
            else:
                if -k <= (dict[nums[index]]-index) <= k:
                    return True #find the duplicate
                dict[nums[index]] =index #update the index
            index += 1
        
        return False